### PR TITLE
Add product query support for Product Summary block

### LINKED ISSUE #7334: Product Query: [Atomic Block] Convert Product Summary Block
Refactor the Product Summary block to support the new Product Query block.

Requirements:

- Must be client-side rendered in the editor.
- Must be server-side rendered (with hydration support) on the frontend.
- The block should be kept independent from the Query Loop block for use with future blocks - see the Single Product block as an example.

### PR DESCRIPTION
**Pull request by imanish003**
Adds Product Query support for the atomic Product Summary block

On the client side, when the Product Summary block is used within the product query block, the markup will be rendered on the server side - No javascript related to Product Summary block will be rendered.

Fixes #7334

### Testing
1. Checkout this branch and run npm start
2. Be sure that you have the latest version of the Gutenberg feature plugin installed and activated
3. Create a post/page and add the Product Query block.
4. Add Product Summary block to Post Template of product query block as shown in video below:

https://user-images.githubusercontent.com/16707866/204458088-0e719130-d858-4573-88b5-99d4922513ab.mov

5. Now save the post & confirm that: 
    1. No JS related to the Product Summary block is loaded (search in your network tab for product-summary-frontend.js). As you can see in the screenshot below, product-summary-frontend.js is showing. **This shouldn't be visible in the network tab.**
image
    2. UI of block should be identical in Frontend & Editor.
    3. The settings are reflected correctly (eg: custom style). Try changing custom style from the sidebar.
    4. ADDITIONAL CSS CLASS(ES)(available in advanced toggle in sidebar) should be added to the container div
    5. Summary text styling set using WYSIWYG editor should work in Editor & frontend. By WYSIWYG editor, I mean following editing block of product:
image

* [X] Do not include in the Testing Notes 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [ ] Feature plugin
* [X] Experimental

### CODE REVIEW

**Comment by sunyatasattva** [comment]
Curious here: why skip changelog? I think this is changelog-worthy.

File: src/BlockTypes/ProductSummary.php [hunk 1]
```diff
@@ -51,5 +53,67 @@ protected function get_block_type_supports() {
...
+		}
+
+		$classes_and_styles = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
+		$classname          = isset( $attributes['className'] ) ? $attributes['className'] : '';
+		$style              = isset( $classes_and_styles['styles'] ) ? ' style="' . $classes_and_styles['styles'] . '" ' : '';
+
+		$output  = '<div class="wc-block-components-product-summary ' . $classes_and_styles['classes'] . ' ' . $classname . '"';
+		$output .= $style;
+		$output .= '>';
+		$output .= '<p>' . $summary_content . '</p>';
```
Comment (Aljullu) [inline src/BlockTypes/ProductSummary.php]:
```suggestion
		$output .= '<p>' . wp_kses_post( $summary_content ) . '</p>';
```

File: src/BlockTypes/ProductSummary.php [hunk 2]
```diff
@@ -51,5 +53,67 @@ protected function get_block_type_supports() {
...
+
+		if ( ! $summary_content ) {
+			return null;
+		}
+
+		$classes_and_styles = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
+		$classname          = isset( $attributes['className'] ) ? $attributes['className'] : '';
+		$style              = isset( $classes_and_styles['styles'] ) ? ' style="' . $classes_and_styles['styles'] . '" ' : '';
+
+		$output  = '<div class="wc-block-components-product-summary ' . $classes_and_styles['classes'] . ' ' . $classname . '"';
```
Comment (Aljullu) [inline src/BlockTypes/ProductSummary.php]:
```suggestion
		$output  = '<div class="wc-block-components-product-summary ' . esc_attr( $classes_and_styles['classes'] ) . ' ' . esc_attr( $classname ) . '"';
```

File: src/BlockTypes/ProductSummary.php [hunk 3]
```diff
@@ -51,5 +53,67 @@ protected function get_block_type_supports() {
...
+		$output .= $style;
```
Comment (Aljullu) [inline src/BlockTypes/ProductSummary.php]: I would inline the ternary from L91 and add `esc_atr()` around `$classes_and_styles['styles']`.

File: src/BlockTypes/ProductSummary.php [hunk 4]
```diff
@@ -51,5 +53,67 @@ protected function get_block_type_supports() {
...
+	 */
+	protected function get_summary_content( $product ) {
+		$short_description = $product->get_short_description();
+		$summary_content   = $short_description ? $short_description : $product->get_description();
+
+		if ( ! $summary_content ) {
+			return null;
+		}
+
+		$summary = wc_trim_string( $summary_content, 150, '...' );
```
Comment (Aljullu) [inline src/BlockTypes/ProductSummary.php]: [`...`  is the default suffix](https://github.com/woocommerce/woocommerce/blob/410f06b8c8098c9c6be8182c35257745c714308b/plugins/woocommerce/includes/wc-formatting-functions.php#L1082), so I would probably not pass it and rely on the default.
```suggestion
		$summary = wc_trim_string( $summary_content, 150 );
```

**Review by Aljullu — COMMENTED** [review]
Thanks for creating this PR, @imanish003! This is testing well and code overall looks good. However, I think we should escape all values we print in `render()`.

Even when they come from a source we know (ie: `StyleAttributesUtils::get_classes_and_styles_by_attributes()`), we should be escaping everything we print in the HTML. Otherwise, we might introduce a filter in `get_classes_and_styles_by_attributes()` in the future, and a plugin would be able to modify the output and print some malicious code in there.

I saw #7709, #7675 and #7734 are also affected by this. Will you be able to update them (if the PR is open) or create a new PR fixing it (if the original PR has already been merged?). We would need this to be fixed before the release next Monday.

File: src/BlockTypes/ProductSummary.php (hunk 4 above)
Comment (imanish003) [reply src/BlockTypes/ProductSummary.php]: Nice catch! I will remove it 🤝

**Comment by imanish003** [comment]
Hi @Aljullu,
Thanks for pointing out the escaping issue 🙏🏻 I have made the changes. Can you please have a look again? Meanwhile, I am going to make the same changes in PRs too 🙂

File: src/BlockTypes/ProductSummary.php [hunk 5]
```diff
@@ -51,5 +53,67 @@ protected function get_block_type_supports() {
...
+			$this->register_chunk_translations( [ $this->block_name ] );
+			return $content;
+		}
+
+		$post_id         = $block->context['postId'];
+		$product         = wc_get_product( $post_id );
+		$summary_content = self::get_summary_content( $product );
+
+		if ( ! $summary_content ) {
+			return null;
```
Comment (Aljullu) [inline src/BlockTypes/ProductSummary.php]: Looking at [this function](https://github.com/WordPress/wordpress-develop/blob/0cb8475c0d07d23893b1d73d755eda5f12024585/src/wp-includes/blocks.php#L980-L990), I think the `render()` function should return a string, so this should probably be:
```suggestion
			return '';
```

File: src/BlockTypes/ProductSummary.php [hunk 6]
```diff
@@ -51,5 +53,67 @@ protected function get_block_type_supports() {
...
+		}
+
+		$classes_and_styles = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
+		$classname          = isset( $attributes['className'] ) ? $attributes['className'] : '';
+
+		$output  = '<div class="wc-block-components-product-summary '
+			. esc_attr( $classes_and_styles['classes'] ) . ' ' . esc_attr( $classname ) . '"'
+			. isset( $classes_and_styles['styles'] ) ? ' style="' . esc_attr( $classes_and_styles['styles'] ) . '"' : '';
```
Comment (Aljullu) [inline src/BlockTypes/ProductSummary.php]: The tertiary like this doesn't seem to work well for me (sorry, I know if was my original suggestion). :sweat: 

imatge

But I think it you do it as a normal if, it works well, then:

```PHP
if ( isset( $classes_and_styles['styles'] ) ) {
	$output .= ' style="' . esc_attr( $classes_and_styles['styles'] ) . '"';
}
```

**Review by Aljullu — COMMENTED** [review]
Thanks for updating this PR, @imanish003! I had a small issue when the block had some custom styling, but besides that, everything looks good. I commented it inline below.

**Comment by Aljullu** [comment]
> Curious here: why skip changelog? I think this is changelog-worthy.

@sunyatasattva I guess it's because the Product Query block hasn't been released yet (until next release), so probably there is no need to mention all its improvements in the changelog until it has been published, no?

**Comment by sunyatasattva** [comment]
> > Curious here: why skip changelog? I think this is changelog-worthy.
> 
> @sunyatasattva I guess it's because the Product Query block hasn't been released yet (until next release), so probably there is no need to mention all its improvements in the changelog until it has been published, no?

Yep, @Aljullu, we talked privately with Manish and decided that from next release onwards, these kind of changes should have changelog entries, but for now it's ok like this.

The downside is that, as we release, these won't appear on the list. But I guess our Release Post can cover everything.

File: src/BlockTypes/ProductSummary.php (hunk 5 above)
Comment (imanish003) [reply src/BlockTypes/ProductSummary.php]: Nice catch. I will update it. Thanks for pointing this out 🙂

File: src/BlockTypes/ProductSummary.php (hunk 6 above)
Comment (imanish003) [reply src/BlockTypes/ProductSummary.php]: Oh, I did test the PR after making the changes, but it looks like my changes weren't reflected correctly while I was doing the testing. Thanks for pointing this out. 🙂

**Review by Aljullu — APPROVED** [review]
This is testing well and code looks good. Thanks @imanish003! :ship:

**Pull request merged**